aea/manager.py

```diff
@@ -0,0 +1,534 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2020 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the implementation of AEA agents manager."""
+import asyncio
+import copy
+import os
+import threading
+from asyncio.tasks import FIRST_COMPLETED
+from shutil import rmtree
+from threading import Thread
+from typing import Callable, Dict, List, Optional, Set
+
+from aea.aea import AEA
+from aea.aea_builder import AEABuilder
+from aea.cli.registry.fetch import fetch_agent
+from aea.cli.utils.context import Context
+from aea.configurations.base import ComponentId, PublicId
+from aea.configurations.constants import DEFAULT_LEDGER
+from aea.crypto.helpers import create_private_key
+from aea.helpers.base import yaml_load_all
+
+
+class Project:
+    """Agent project representation."""
+
+    def __init__(self, public_id: PublicId, path: str):
+        """Init project with public_id and project's path."""
+        self.public_id = public_id
+        self.path = path
+        self.agents: Set[str] = set()
```

Comment:
Not clear from current implementation why we need `agents` ...I know we do but seems you're missing some code?

Reply:
...I saw it further below.